hastory-digest/src/Data/Hastory/Types/Digest.hs

```diff
@@ -22,3 +29,13 @@ instance HashAlgorithm a => PersistField (Digest a) where
 
 instance HashAlgorithm a => PersistFieldSql (Digest a) where
   sqlType _ = sqlType (Proxy :: Proxy B.ByteString)
+
+instance FromJSON (Digest SHA256) where
+  parseJSON =
+    withText "Digest SHA256" $ \text -> do
+      let failureMsg err = fail ("Failed to parse (Digest SHA256): " ++ T.unpack err)
+      bytestring <- either failureMsg pure (Base64.decodeBase64 $ T.encodeUtf8 text)
+      maybe (failureMsg text) pure (digestFromByteString bytestring)
```

Comment:
This is quite complex, and unnecessarily so. I prefer case expressions but it's up to you.

Reply:
Readability is very important to me. I've changed it to case expressions.